Enable the use of non-origin oAuth urls + new window options

jookyboi opened the pull request:
I have an Angular2 app which uses a separate Rails/Devise backend. I needed the oAuth flow to hit that server rather than the host which serves my frontend client. This pull helps deal with that.

- Added new `oAuthBase` option for the oAuth flow
- Added new `apiBase` option for other API calls
- Added a [string:string] object `oAuthWindowOptions` to allow users to set additional options on the new window (ex. `oAuthWindowOptions: { width: '800', height: '600' }`)
- I switched to using backticks for strings which have variable interpolation. It didn't compile with just straight up single-quotes for me
- Updated docs

Thanks for taking the lead on an excellent package.

neroniaky commented:
I think so too. Didn't have time to test it yet.

hrangel commented:
@neroniaky that will even fix the issue I mentioned in #119 , the issue is with the single quote chars when building the oauth path, it should be ` char so the literal string replacing works.

jookyboi commented:
Thanks. Since the PR was opened, I also added an optional `resetPasswordToken` to `updatePassword()` so that users going through the Devise reset-password flow could update their password.

I added an optional `name?` as well to the `RegisterData` model since I needed it for display.

neroniaky commented:
Thanks @jookyboi!

jookyboi commented:
👍
